tardis/io/logger/logger.py

```diff
@@ -1,165 +1,265 @@
+logger_widget = pn.Tabs(
+    *[(title, log_outputs[title]) for title in tab_order],
+    height=350,
+    sizing_mode='stretch_width'
+)
 
-class FilterLog:
-    """
-    Filter Log Class for Filtering Logging Output
-    to a particular level
 
-    Parameters
-    ----------
-    log_level : logging object
-        allows to have a filter for the
-        particular log_level
-    """
+@dataclass
+class LoggingConfig:
+    LEVELS: dict[str, int] = field(default_factory=lambda: {
+        "NOTSET": logging.NOTSET,
+        "DEBUG": logging.DEBUG,
+        "INFO": logging.INFO,
+        "WARNING": logging.WARNING,
+        "ERROR": logging.ERROR,
+        "CRITICAL": logging.CRITICAL,
+    })
 
-    def __init__(self, log_level):
-        self.log_level = log_level
+    COLORS: dict[int | str, str] = field(default_factory=lambda: {
+        logging.INFO: "#D3D3D3",
+        logging.WARNING: "orange",
+        logging.ERROR: "red",
+        logging.CRITICAL: "orange",
+        logging.DEBUG: "blue",
+        "default": "black",
+    })
+    
+    DEFAULT_LEVEL = "INFO"
+    DEFAULT_SPECIFIC_STATE = False
 
-    def filter(self, log_record):
-        """
-        filter() allows to set the logging level for
-        all the record that are being parsed & hence remove those
-        which are not of the particular level
-
-        Parameters
-        ----------
-        log_record : logging.record
-            which the paricular record upon which the
-            filter will be applied
-
-        Returns
-        -------
-        boolean : True, if the current log_record has the
-            level that of the specified log_level
-            False, if the current log_record doesn't have the
-            same log_level as the specified one
-        """
-        return log_record.levelno == self.log_level
-
-
-def logging_state(log_level, tardis_config, specific_log_level):
-    """
-    Function to set the logging configuration for the simulation output
-    Called from within run_tardis()
-    Configured via functional arguments passed through run_tardis() - log_level & specific_log_level
-    Configured via YAML parameters under `debug` section - log_level & specific_log_level
-
-    Parameters
-    ----------
-    log_level: str
-        Allows to input the log level for the simulation
-        Uses Python logging framework to determine the messages that will be output
-    specific_log_level: boolean
-        Allows to set specific logging levels. Logs of the `log_level` level would be output.
-    """
-    if "debug" in tardis_config:
-        specific_log_level = (
-            tardis_config["debug"]["specific_log_level"]
-            if specific_log_level is None
-            else specific_log_level
-        )
 
-        logging_level = (
-            log_level if log_level else tardis_config["debug"]["log_level"]
-        )
+LOGGING_LEVELS = LoggingConfig().LEVELS
+
+class TardisLogger:
```

Comment:
should be `TARDISLogger`

Reply:
Fixed in https://github.com/tardis-sn/tardis/pull/2872/commits/853dea41831b06ac11ec3a2c43ad8e9974d91ffe